### PR TITLE
🦋 Implement empty state in Applications index

### PR DESCRIPTION
**Pull request by josemigallas**
* Service: https://issues.redhat.com/browse/THREESCALE-9797
* Audience: https://issues.redhat.com/browse/THREESCALE-9970
* Buyers: https://issues.redhat.com/browse/THREESCALE-9965

Before:
Screenshot 2024-06-19 at 16 00 04

After:
Screenshot 2024-06-19 at 17 31 04
Screenshot 2024-06-19 at 17 30 24
Screenshot 2024-06-19 at 17 30 11

And also empty search state:
Screenshot 2024-06-19 at 17 36 15

### CODE REVIEW

File: app/controllers/buyers/applications_controller.rb [hunk 1]
```diff
@@ -36,6 +35,10 @@ def create
 
   protected
 
+  def accessible_not_bought_cinstances
```
Comment (mayorova) [inline app/controllers/buyers/applications_controller.rb:38]: This naming is quite confusing... :grimacing:  Especially looking at the implementation. I know that it was already like this. But dunno, maybe we could rename? Something like `accessible_buyers_cinstances` to reflect that it represents the apps that were bought by the buyer (developer), and not the provider itself (which is, as I understand, the meaning of this).

Comment (mayorova) [reply app/controllers/buyers/applications_controller.rb:38]: Also, this added scope is new (and not moved from somewhere), and I was wondering - how did it use to filter? I guess by the `define_search_scope` method? So, I'm wondering if these two are redundant now.

And the same actually for the service controller.

**Comment by mayorova** [comment]
Works well :+1: Some minor comments on the code, but OK to keep it as it is.

**Review by lvillen — APPROVED** [review]
Works well for me too!

File: app/controllers/buyers/applications_controller.rb (hunk 1 above)
Comment (josemigallas) [reply app/controllers/buyers/applications_controller.rb:38]: I couldn't figure out what "accessible_not_bought_cinstances" meant but if you think your suggestion makes more sense I'm all for changing it.

About the scope thingy, the apps under "account" scope buyer/accounts/applications or whatever is filtered by injecting the account id as a hidden parameter, which is very weird IMO. I noticed this `super.bought_by(@account)` in some other controller (or it was maybe in the model) and I thought it is the "right" way to do it.

**Review by lvillen — APPROVED** [review]

**Pull request merged**